lib/Utils/Polynomial/ChebyshevPatersonStockmeyer.h

```diff
@@ -22,6 +22,65 @@ namespace polynomial {
 // evaluation of the Chebyshev polynomial.
 constexpr double kMinCoeffs = 1e-15;
 
+// A C++ port of Lattigo's `bignum.OptimalSplit` function.
+// This function determines the optimal split point for the Paterson-Stockmeyer
+// algorithm by calculating the number of multiplications for two possible
+// split points and choosing the better one.
+//
+// Args:
+//   logDegree: ceil(log2(degree)) of the polynomial.
+//
+// Returns:
+//   The optimal split parameter (log scale).
```

Comment:
```suggestion
//   The optimal split parameter (log split).
```
? actually not sure what log scale meant